packages/atomic/src/components/atomic-breadcrumb-manager/atomic-breadcrumb-manager.tsx

```diff
@@ -45,51 +61,267 @@ export class AtomicBreadcrumbManager {
+        >
+          {breadcrumbValue}
+        </button>
+      </li>
+    ));
+    return [...renderedBreadcrumbs, this.mainClear];
   }
 
   render() {
+    if (this.hasActiveBreadcrumbs()) {
+      return (
+        <div class="row">
+          <span class=" col-9">
+            {this.facetBreadcrumbs}
+            {this.numericFacetBreadcrumbs}
+            {this.dateFacetBreadcrumbs}
+            {this.categoryFacetBreadcrumbs}
+          </span>
+          <span class="col-3 text-right">
+            {this.getClearAllFiltersButton()}
+          </span>
+        </div>
+      );
+    }
+  }
+
+  private getClearAllFiltersButton() {
     return (
-      <div class="breadcrumb-manager">
-        {this.facetBreadcrumbs}
-        {this.numericFacetBreadcrumbs}
-        {this.dateFacetBreadcrumbs}
-        {this.categoryFacetBreadcrumbs}
-      </div>
+      <button
+        part="breadcrumb-button"
+        class={this.buttonClasses}
+        onClick={() => {}} //TODO https://coveord.atlassian.net/browse/KIT-269
+      >
+        Clear All Filters
+      </button>
     );
   }
+
+  private showFacetCollapsedBreadcrumbs(field: string) {
+    this.collapsedBreadcrumbsState = {
+      ...this.collapsedBreadcrumbsState,
+      [field]: false,
+    };
+  }
+
+  private collapsedBreadcrumbsHandler<T extends BaseFacetValue>(
+    breadcrumb: Breadcrumb<T>
+  ) {
+    let breadcrumbsToShow = [];
+    let moreButton = undefined;
+
+    if (!this.collapsedBreadcrumbsState[breadcrumb.field]) {
+      breadcrumbsToShow = breadcrumb.values;
+      breadcrumbsToShow.length <= this.collapseThreshold
+        ? (this.collapsedBreadcrumbsState[breadcrumb.field] = true)
+        : null;
+    } else {
+      breadcrumbsToShow = breadcrumb.values.slice(0, this.collapseThreshold);
+      const collapsedBreadcrumbNumber =
+        breadcrumb.values.length - this.collapseThreshold;
+
+      if (collapsedBreadcrumbNumber > 0) {
+        moreButton = (
+          <li part="breadcrumb-value">
+            <button
+              part="breadcrumb-button"
+              class={this.buttonClasses}
+              onClick={() =>
+                this.showFacetCollapsedBreadcrumbs(breadcrumb.field)
+              }
+            >
+              {collapsedBreadcrumbNumber} more...
+            </button>
+          </li>
+        );
+      }
+    }
+    return {breadcrumbsToShow, moreButton};
+  }
+
+  private hasActiveBreadcrumbs() {
+    return !this.isEmpty([
+      ...this.state.facetBreadcrumbs.filter(
+        (breadcrumb) => !this.isEmpty(breadcrumb.values)
+      ),
+      ...this.state.numericFacetBreadcrumbs.filter(
+        (breadcrumb) => !this.isEmpty(breadcrumb.values)
+      ),
+      ...this.state.dateFacetBreadcrumbs.filter(
+        (breadcrumb) => !this.isEmpty(breadcrumb.values)
+      ),
+      ...this.state.categoryFacetBreadcrumbs.filter(
+        (breadcrumb) => !this.isEmpty(breadcrumb.path)
+      ),
+    ]);
+  }
+
+  private isEmpty(array: any[]) {
```

Comment:
Normally we should try to not use `any` if possible. Since your not using the content, but only looking at the length, `unknown[]` should be enough.
Also nothing  wrong with directly looking at the length, it's not more/less code e.g.
```
(breadcrumb) => !!breadcrumb.values.length
```

I would move this hasActiveBreadcrumbs in the `headless-breadcrumb-manager`'s state directly instead of here in atomic